component/loki/source/file/file.go

```diff
@@ -0,0 +1,277 @@
+package file
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/go-kit/log/level"
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/discovery"
+	"github.com/grafana/loki/clients/pkg/promtail/api"
+	"github.com/grafana/loki/clients/pkg/promtail/positions"
```

Comment:
Do we plan to replace these with agent code (probably a fork of Promtail) at some point in a future PR?

Reply:
Yes, the plan is to start porting over everything, maybe as well do it right away, right?

The `api` package is a single file which multiple components will reuse and while the `positions` package is a little larger, it's not super complicated by any means.

Reply:
I've brought both packages over under `component/common` and am using our own copies for the new component.